libcxx/include/__memory/temporary_buffer.h

```diff
@@ -22,8 +22,8 @@
 _LIBCPP_BEGIN_NAMESPACE_STD
 
 template <class _Tp>
-_LIBCPP_NODISCARD _LIBCPP_HIDE_FROM_ABI _LIBCPP_NO_CFI _LIBCPP_DEPRECATED_IN_CXX17 pair<_Tp*, ptrdiff_t>
-get_temporary_buffer(ptrdiff_t __n) _NOEXCEPT {
+_LIBCPP_NODISCARD _LIBCPP_HIDE_FROM_ABI _LIBCPP_NO_CFI pair<_Tp*, ptrdiff_t>
+__get_temporary_buffer(ptrdiff_t __n) _NOEXCEPT {
```

Comment:
I don't think adding an internal version is the right call here. We should replace the call sites instead. I'd probably go for a `unique_ptr` and calling `allocator<T>::allocate()`, since that makes it trivial to implement [P2562R1](https://wg21.link/P2562R1).

Reply:
Initially, I though it would be trivial to switch to call `allocator<T>::allocate()`. However, it's probably intented to call `nothrow` versions of `operator new` in this function. And per #69498, I think we want to support reconvering from allocation failure under -fno-exceptions.

Given `allocate` calls throwing versions of `operator new`, it can lead to undesired termination at run time under -fno-exceptions, while such termination is avoided currently.

Reply:
+1 I think we should have an alternative for `get_temporary_buffer` and use that internally.

Reply:
I'm now trying to create constexpr-friendly `__scoped_temporary_buffer`, but the `constexpr` isn't meaningful until [P2562R1](https://wg21.link/P2562R1) gets implemented.